atomate/vasp/workflows/base/magnetism.py

```diff
@@ -0,0 +1,834 @@
+
+        # our magnetically ordered structures will be
+        # stored here once generated and also store which
+        # transformation created them, this is used for
+        # book-keeping/user interest, and
+        # is be a list of strings in ("fm", "afm",
+        # "ferrimagnetic_by_species", "ferrimagnetic_by_motif",
+        # "afm_by_motif", "input_structure")
+        self.ordered_structures, self.ordered_structure_origins = [], []
+
+        formula = structure.composition.reduced_formula
+
+        # to process disordered magnetic structures, first make an
+        # ordered approximation
+        if not structure.is_ordered:
+            raise ValueError(
+                "Please obtain an ordered approximation of the "
+                "input structure ({}).".format(formula)
+            )
+
+        # CollinearMagneticStructureAnalyzer is used throughout:
+        # it can tell us whether the input is itself collinear (if not,
+        # this workflow is not appropriate), and has many convenience
+        # methods e.g. magnetic structure matching, etc.
+        self.input_analyzer = CollinearMagneticStructureAnalyzer(
+            structure, default_magmoms=default_magmoms, overwrite_magmom_mode="none"
+        )
+
+        # this workflow enumerates structures with different combinations
+        # of up and down spin and does not include spin-orbit coupling:
+        # if your input structure has vector magnetic moments, this
+        # workflow is not appropriate
+        if not self.input_analyzer.is_collinear:
+            raise ValueError("Input structure ({}) is non-collinear.".format(formula))
+
+        self.sanitized_structure = self._sanitize_input_structure(structure)
+
+        # we will first create a set of transformations
+        # and then apply them to our input structure
+        self.transformations = self._generate_transformations(self.sanitized_structure)
+        self._generate_ordered_structures(
+            self.sanitized_structure, self.transformations
+        )
+
+    def _sanitize_input_structure(self, input_structure):
+        """
+        Sanitize our input structure by removing magnetic information
+        and making primitive.
+        """
+
+        input_structure = input_structure.copy()
+
+        # remove any annotated spin
+        input_structure.remove_spin()
+
+        # sanitize input structure: first make primitive ...
+        input_structure = input_structure.get_primitive_structure(use_site_props=False)
+
+        # ... and strip out existing magmoms, which can cause conflicts
+        # with later transformations otherwise since sites would end up
+        # with both magmom site properties and Specie spins defined
+        if "magmom" in input_structure.site_properties:
+            input_structure.remove_site_property("magmom")
+
+        return input_structure
+
+    def _generate_transformations(self, structure):
```

Comment:
I know this is an internal method, but better docstring about input and output parameters would help

Reply:
Ok, will add